File: Sorting_And_Searching_In_Java/7. Doubt Clearing Session/LargestNumber.java
import java.util.Arrays;

public class LargestNumber {
	public static void main(String[] args) {
		System.out.println(largestNumber(new int[] {3, 30, 34, 5, 9}));
	}
	
	public static String largestNumber(int[] nums) {
		String[] array = new String[nums.length];
		for(int i = 0; i < nums.length; i++) {
			array[i] = String.valueOf(nums[i]);
		}
		
		//a="100", b = "90"  "90100" > "10090"
		Arrays.sort(array, (a, b) -> (b + a).compareTo(a + b));
		if(array[0].equals("0")) return "0";
		StringBuilder sb = new StringBuilder();
		for(int i = 0; i < array.length; i++) {
			sb.append(array[i]);
		}
		
		return sb.toString();
	}
}

File: Pulkit Chhabra/Greedy Algorithms (April 2021)/L2 - Exchange Argument and Activity Selection/largest_number.cpp
 bool cmp(int a,int b)
{
    return to_string(a) + to_string(b) > to_string(b) + to_string(a);
}


class Solution {
    
    public:
    string largestNumber(vector<int>& nums) {
        
        if(*max_element(nums.begin(),nums.end())==0)
            return "0";
        
        sort(nums.begin(), nums.end(), cmp);
        string ans;
        
        for(int i : nums)
            ans += to_string(i);
        
        return ans;
            
    }
};
